### PR TITLE
ui: Fixed git settings input field repeat bug + cleanup

### PR DESCRIPTION
**Pull request by sabrinako**
**Issue This PR is Fixing**
#1397

### CODE REVIEW

File: ui/app/components/app-form/project-settings.ts [hunk 1]
```diff
@@ -48,24 +48,24 @@ const DEFAULT_PROJECT_MODEL = {
...
         username: '',
-        password: ''
+        password: '',
       },
       ssh: {
         user: '',
         password: '',
         privateKeyPem: '',
-      }
+      },
```
Comment (koesbong) [inline ui/app/components/app-form/project-settings.ts:60]: is this comma needed here?

Comment (koesbong) [reply ui/app/components/app-form/project-settings.ts:60]: hmm, i also see a trailing comma on line 59 and 54 too. is that the code convention?

Comment (gregone) [reply ui/app/components/app-form/project-settings.ts:60]: Yes, the linter enforces a `comma on last line of objects` rule

File: ui/tests/integration/components/app-form/project-settings-test.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    await fillIn('#git-source-username', 'admin');
+    await fillIn('#git-source-password', 'password');
+    await clearRender();
+
+    this.set('project2', {});
+    await render(hbs`<AppForm::ProjectSettings @project={{this.project2}} />`);
+
+    assert.dom('#git-source-url').hasValue('');
+    assert.dom('#git-source-username').hasValue('');
+    assert.dom('#git-source-password').hasValue('');
```
Comment (gregone) [inline ui/tests/integration/components/app-form/project-settings-test.ts:22]: This test is good, but considering the changes above to the default data & `authCase` logic, I'd like to add a test with existing `dataSource` data covering both ssh and basic settings to verify that everything still works. I'm happy to work on adding that test or to pair with you tomorrow on that.

**Review by gregone — CHANGES_REQUESTED** [review]

File: ui/app/components/app-form/project-settings.ts [hunk 3]
```diff
@@ -14,27 +14,27 @@ const FORMAT = {
 };
 class ProjectModel {
   name: string;
-  applicationsList: [];
+  applicationsList: never[];
```
[the remark this answers is not in the document]
Comment (sabrinako) [reply ui/app/components/app-form/project-settings.ts:18]: This suppressed a TS error we received when trying to make `DEFAULT_PROJECT_MODEL` an instance of `ProjectModel`. With the new JSON approach this error doesn't seem to occur, so I'll remove the `never` 👍

**Review by gregone — APPROVED** [review]

**Pull request merged**